Sources/SwiftParserDiagnostics/MissingNodesError.swift

```diff
@@ -331,15 +392,20 @@ extension ParseDiagnosticsGenerator {
     }
 
     let changes = missingNodes.enumerated().map { (index, missingNode) -> FixIt.Changes in
-      if index == 0, let token = missingNode.as(TokenSyntax.self), token.tokenKind.isPunctuation == true, token.previousToken(viewMode: .sourceAccurate)?.tokenKind.isPunctuation == false {
+      let previousTokenKind = missingNode.previousToken(viewMode: .sourceAccurate)?.tokenKind
+      if index == 0,
+         let token = missingNode.as(TokenSyntax.self),
+         (token.tokenKind.isPunctuation && previousTokenKind?.isPunctuation == false
+          || !token.tokenKind.isPunctuation && previousTokenKind?.isPunctuation == true) {
```

Comment:
Style idea (again not sure if you agree)

```suggestion
         let previousTokenIsPunctionation = previousTokenKind?.isPunctuation,
         token.tokenKind.isPunctuation != previousTokenIsPunctuation) {
```

You could even inline `previousTokenKind` into the `if`

Reply:
Uh... yeah, I can just pull `previousTokenKind` in. Wonder how late it was when I wrote this 😆. Thanks, good suggestion!

Reply:
@ahoppen I ended up having to change this. This broke eg. required keywords following a `{`. I now specifically check if the current token is a punctuation and the next token is an identifier. That would also break if we ever had a required identifier after `{` but I don't think that ever happens? Open to other ideas here, this seems pretty hacky right now.

Reply:
Sounds reasonable to me.